### PR TITLE
add endpoint spec to get events from a test run

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR adds the openAPI spec to get events generated from a test run. The model is based on diagram on #2040

```mermaid
classDiagram
    Event <|-- DataStoreConnection
    Event <|-- Polling
    Event <|-- Output
    Event : string type
    Event : enum[trigger, trace, test] stage
    Event : string description
    Event : date created_at
    Event : string test_id
    Event : string run_id

    class DataStoreConnection{
      DataStoreTestConnection info
    }

    class Polling{
      int number_of_spans
      int iteration_number
      string reason_of_next_iteration
      boolean is_complete
    }

    class Output{
      string warning
      string error
      string output_name
    }
```

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: api/testEvents.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          $ref: "#/components/schemas/PollingInfo"
+        outputs:
+          type: array
+          items:
+            $ref: "#/components/schemas/OutputInfo"
+
+    PollingInfo:
+      type: object
+      properties:
+        numberSpans:
```
[the remark this answers is not in the document]
Comment (mathnogueira) [reply api/testEvents.yaml:36]: we could add a `type` field and then add objects to represent each type, would that be enough?

File: api/testEvents.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          type: boolean
+
+    OutputInfo:
+      type: object
+      properties:
+        logLevel:
+          type: string
+          enum:
+            - warning
+            - error
```
Comment (jorgeepc) [inline api/testEvents.yaml:52]: Should we include a `info` type?

File: api/testEvents.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (schoren) [reply api/testEvents.yaml:36]: makes sense, that's the approach we took on all similar situations

File: api/testEvents.yaml (hunk 2 above)
Comment (mathnogueira) [reply api/testEvents.yaml:52]: I don't think it's worth unless we want to show in the event log what are the values for each output after they were resolved. @xoscar what do you think?

**Comment by xoscar** [comment]
Actually, can we consider adding the specs for this as well? https://github.com/kubeshop/tracetest/issues/2183

File: api/testEvents.yaml (hunk 2 above)
Comment (xoscar) [reply api/testEvents.yaml:52]: I think the info type is not something we'll be using in the near future so I think its fine to not have it

Comment (jorgeepc) [reply api/testEvents.yaml:59]: Yeah, makes sense @mathnogueira

File: api/testEvents.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mathnogueira) [reply api/testEvents.yaml]: Done

**Review by jorgeepc — APPROVED** [review]

**Comment by mathnogueira** [comment]
> Actually, can we consider adding the specs for this as well? #2183

added it

**Review by schoren — APPROVED** [review]

**Review by danielbdias — APPROVED** [review]

**Pull request merged**